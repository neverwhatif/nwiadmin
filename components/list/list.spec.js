import React from 'react';
import { shallow } from 'enzyme';

import List from './index';

jest.mock('./listitem', () => 'ListItem');

const data = [{
    id: 1,
    title: 'Lorem',
    description: 'lorem',
}, {
    id: 2,
    title: 'Ipsum',
    description: 'ipsum',
}];

const mockTransformer = item => ({
    id: item.id * 2,
});

describe('List', () => {
    const component = shallow(<List data={data} />);
    const componentIsDisabled = shallow(<List data={data} isDisabled />);
    const componentWithNoData = shallow(<List data={[]} />);
    const componentWithTransformer = shallow(<List data={data} transformer={mockTransformer} />);
    const componentWithPreload = shallow(<List data={data} shouldInitPreload />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentIsDisabled).toMatchSnapshot();
        expect(componentWithNoData).toMatchSnapshot();
        expect(componentWithTransformer).toMatchSnapshot();
        expect(componentWithPreload).toMatchSnapshot();
    });
    it('should render list items', () => {
        expect(component.find('ListItem').length).toBe(2);
    });
    it('should render disabled', () => {
        expect(componentIsDisabled.hasClass('rootDisabled')).toBe(true);
    });
    it('should render nothing if no data', () => {
        expect(componentWithNoData.type()).toBeNull();
    });
    it('should have a default transformer', () => {
        expect(List.defaultProps.transformer('lorem')).toBe('lorem');
    });
    it('should have a transformer', () => {
        expect(componentWithTransformer.find('li').first().key()).toBe('2');
    });
    it('should pass shouldInitPreload', () => {
        expect(component.find('ListItem').first().prop('shouldInitPreload')).toBe(false);
        expect(componentWithPreload.find('ListItem').first().prop('shouldInitPreload')).toBe(true);
    });
});
