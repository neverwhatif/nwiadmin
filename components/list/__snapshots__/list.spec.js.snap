// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    columns={Array []}
    data={
      Array [
        Object {
          "description": "lorem",
          "id": 1,
          "title": "Lorem",
        },
        Object {
          "description": "ipsum",
          "id": 2,
          "title": "Ipsum",
        },
      ]
    }
    isDisabled={false}
    shouldInitPreload={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <ListItem
            description="lorem"
            id={1}
            shouldInitPreload={false}
            title="Lorem"
          />
        </li>,
        <li>
          <ListItem
            description="ipsum"
            id={2}
            shouldInitPreload={false}
            title="Ipsum"
          />
        </li>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <ListItem
            description="lorem"
            id={1}
            shouldInitPreload={false}
            title="Lorem"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cta": undefined,
            "description": "lorem",
            "id": 1,
            "shouldInitPreload": false,
            "title": "Lorem",
          },
          "ref": null,
          "rendered": null,
          "type": "ListItem",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <ListItem
            description="ipsum"
            id={2}
            shouldInitPreload={false}
            title="Ipsum"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cta": undefined,
            "description": "ipsum",
            "id": 2,
            "shouldInitPreload": false,
            "title": "Ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": "ListItem",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <ListItem
              description="lorem"
              id={1}
              shouldInitPreload={false}
              title="Lorem"
            />
          </li>,
          <li>
            <ListItem
              description="ipsum"
              id={2}
              shouldInitPreload={false}
              title="Ipsum"
            />
          </li>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <ListItem
              description="lorem"
              id={1}
              shouldInitPreload={false}
              title="Lorem"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": undefined,
              "description": "lorem",
              "id": 1,
              "shouldInitPreload": false,
              "title": "Lorem",
            },
            "ref": null,
            "rendered": null,
            "type": "ListItem",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <ListItem
              description="ipsum"
              id={2}
              shouldInitPreload={false}
              title="Ipsum"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": undefined,
              "description": "ipsum",
              "id": 2,
              "shouldInitPreload": false,
              "title": "Ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": "ListItem",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`List should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    columns={Array []}
    data={
      Array [
        Object {
          "description": "lorem",
          "id": 1,
          "title": "Lorem",
        },
        Object {
          "description": "ipsum",
          "id": 2,
          "title": "Ipsum",
        },
      ]
    }
    isDisabled={true}
    shouldInitPreload={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <ListItem
            description="lorem"
            id={1}
            shouldInitPreload={false}
            title="Lorem"
          />
        </li>,
        <li>
          <ListItem
            description="ipsum"
            id={2}
            shouldInitPreload={false}
            title="Ipsum"
          />
        </li>,
      ],
      "className": "root rootDisabled",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <ListItem
            description="lorem"
            id={1}
            shouldInitPreload={false}
            title="Lorem"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cta": undefined,
            "description": "lorem",
            "id": 1,
            "shouldInitPreload": false,
            "title": "Lorem",
          },
          "ref": null,
          "rendered": null,
          "type": "ListItem",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <ListItem
            description="ipsum"
            id={2}
            shouldInitPreload={false}
            title="Ipsum"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cta": undefined,
            "description": "ipsum",
            "id": 2,
            "shouldInitPreload": false,
            "title": "Ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": "ListItem",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <ListItem
              description="lorem"
              id={1}
              shouldInitPreload={false}
              title="Lorem"
            />
          </li>,
          <li>
            <ListItem
              description="ipsum"
              id={2}
              shouldInitPreload={false}
              title="Ipsum"
            />
          </li>,
        ],
        "className": "root rootDisabled",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <ListItem
              description="lorem"
              id={1}
              shouldInitPreload={false}
              title="Lorem"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": undefined,
              "description": "lorem",
              "id": 1,
              "shouldInitPreload": false,
              "title": "Lorem",
            },
            "ref": null,
            "rendered": null,
            "type": "ListItem",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <ListItem
              description="ipsum"
              id={2}
              shouldInitPreload={false}
              title="Ipsum"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": undefined,
              "description": "ipsum",
              "id": 2,
              "shouldInitPreload": false,
              "title": "Ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": "ListItem",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`List should render 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    columns={Array []}
    data={Array []}
    isDisabled={false}
    shouldInitPreload={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`List should render 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    columns={Array []}
    data={
      Array [
        Object {
          "description": "lorem",
          "id": 1,
          "title": "Lorem",
        },
        Object {
          "description": "ipsum",
          "id": 2,
          "title": "Ipsum",
        },
      ]
    }
    isDisabled={false}
    shouldInitPreload={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <ListItem
            id={2}
            shouldInitPreload={false}
          />
        </li>,
        <li>
          <ListItem
            id={4}
            shouldInitPreload={false}
          />
        </li>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <ListItem
            id={2}
            shouldInitPreload={false}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cta": undefined,
            "id": 2,
            "shouldInitPreload": false,
          },
          "ref": null,
          "rendered": null,
          "type": "ListItem",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": <ListItem
            id={4}
            shouldInitPreload={false}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cta": undefined,
            "id": 4,
            "shouldInitPreload": false,
          },
          "ref": null,
          "rendered": null,
          "type": "ListItem",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <ListItem
              id={2}
              shouldInitPreload={false}
            />
          </li>,
          <li>
            <ListItem
              id={4}
              shouldInitPreload={false}
            />
          </li>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <ListItem
              id={2}
              shouldInitPreload={false}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": undefined,
              "id": 2,
              "shouldInitPreload": false,
            },
            "ref": null,
            "rendered": null,
            "type": "ListItem",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": <ListItem
              id={4}
              shouldInitPreload={false}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": undefined,
              "id": 4,
              "shouldInitPreload": false,
            },
            "ref": null,
            "rendered": null,
            "type": "ListItem",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`List should render 5`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    columns={Array []}
    data={
      Array [
        Object {
          "description": "lorem",
          "id": 1,
          "title": "Lorem",
        },
        Object {
          "description": "ipsum",
          "id": 2,
          "title": "Ipsum",
        },
      ]
    }
    isDisabled={false}
    shouldInitPreload={true}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <ListItem
            description="lorem"
            id={1}
            shouldInitPreload={true}
            title="Lorem"
          />
        </li>,
        <li>
          <ListItem
            description="ipsum"
            id={2}
            shouldInitPreload={true}
            title="Ipsum"
          />
        </li>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <ListItem
            description="lorem"
            id={1}
            shouldInitPreload={true}
            title="Lorem"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cta": undefined,
            "description": "lorem",
            "id": 1,
            "shouldInitPreload": true,
            "title": "Lorem",
          },
          "ref": null,
          "rendered": null,
          "type": "ListItem",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <ListItem
            description="ipsum"
            id={2}
            shouldInitPreload={true}
            title="Ipsum"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cta": undefined,
            "description": "ipsum",
            "id": 2,
            "shouldInitPreload": true,
            "title": "Ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": "ListItem",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <ListItem
              description="lorem"
              id={1}
              shouldInitPreload={true}
              title="Lorem"
            />
          </li>,
          <li>
            <ListItem
              description="ipsum"
              id={2}
              shouldInitPreload={true}
              title="Ipsum"
            />
          </li>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <ListItem
              description="lorem"
              id={1}
              shouldInitPreload={true}
              title="Lorem"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": undefined,
              "description": "lorem",
              "id": 1,
              "shouldInitPreload": true,
              "title": "Lorem",
            },
            "ref": null,
            "rendered": null,
            "type": "ListItem",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <ListItem
              description="ipsum"
              id={2}
              shouldInitPreload={true}
              title="Ipsum"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": undefined,
              "description": "ipsum",
              "id": 2,
              "shouldInitPreload": true,
              "title": "Ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": "ListItem",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
